apps/dashboard/src/components/workflow-editor/steps/configure-step-template-form.tsx

```diff
@@ -35,16 +35,20 @@ export const ConfigureStepTemplateForm = (props: ConfigureStepTemplateFormProps)
 
   const schema = useMemo(() => buildDynamicZodSchema(step.controls.dataSchema ?? {}), [step.controls.dataSchema]);
 
-  const defaultValues = useMemo(
-    () => merge(buildDefaultValues(step.controls.uiSchema ?? {}), step.controls.values),
-    []
-  );
+  const defaultValues = useMemo(() => {
+    // Use the UI Schema to build the default values if it exists else use the data schema (code-first approach) values
+    if (Object.keys(step.controls.uiSchema ?? {}).length !== 0) {
+      return merge(buildDefaultValues(step.controls.uiSchema ?? {}), step.controls.values);
+    }
+
+    return merge(buildDefaultValuesOfDataSchema(step.controls.dataSchema ?? {}), step.controls.values);
+  }, []);
```

Comment:
This updates the form to have the default control values from the code-first step controls if available